source/common/stream_info/stream_info_impl.h

```diff
@@ -238,12 +240,10 @@ struct StreamInfoImpl : public StreamInfo {
 
   uint64_t responseFlags() const override { return response_flags_; }
 
-  void setRouteName(absl::string_view route_name) override {
-    route_name_ = std::string(route_name);
+  const std::string& getRouteName() const override {
+    return route_ != nullptr ? route_->routeName() : EMPTY_STRING;
```

Comment:
@wbpcode I think the route_ is never set for direct response action? It looks like it's only set in the upstream request which is not happening for direct response.

Reply:
Nope, the route_ will be set when we refresh cached route. See

![image](https://github.com/envoyproxy/envoy/assets/12389633/cd9f556f-2c0e-41d4-ae72-1a500c904379)

Reply:
did you encounter some problem in accutally usage?

Reply:
@wbpcode it broken istio's CI, see https://github.com/istio/proxy/pull/4932.

Reply:
Could you provide more speicifc log or problem or context? It's a little hard for me to figure out the root cause or problem.

Reply:
It breaks the implementation of stackdrive filter, @kyessenov can you share the details?

Reply:
I think this PR fixed some bug actually where `route_name` was not populated with RBAC. Now it's populated, and we can update our log entry samples. Not sure if this deserves a notice - it's improving telemetry overall.